reactor-netty-core/src/main/java/reactor/netty/transport/NameResolverProvider.java

```diff
@@ -0,0 +1,591 @@
+		 * the DNS server and will use this max time to live.
+		 * Default to {@link Build#DEFAULT_CACHE_MAX_TIME_TO_LIVE}.
+		 *
+		 * @param cacheMaxTimeToLive the maximum time to live (resolution: seconds)
+		 * @return {@code this}
+		 */
+		NameResolverSpec cacheMaxTimeToLive(Duration cacheMaxTimeToLive);
+
+		/**
+		 * Sets the min time to live of the cached DNS resource records (resolution: seconds).
+		 * If the time to live of the DNS resource record returned by the DNS server is less
+		 * than this min time to live, this resolver will ignore the time to live from
+		 * the DNS server and will use this min time to live.
+		 * Default to {@link Build#DEFAULT_CACHE_MIN_TIME_TO_LIVE}.
+		 *
+		 * @param cacheMinTimeToLive the minimum time to live (resolution: seconds)
+		 * @return {@code this}
+		 */
+		NameResolverSpec cacheMinTimeToLive(Duration cacheMinTimeToLive);
+
+		/**
+		 * Sets the time to live of the cache for the failed DNS queries (resolution: seconds).
+		 * Default to {@link Build#DEFAULT_CACHE_NEGATIVE_TIME_TO_LIVE}.
+		 *
+		 * @param cacheNegativeTimeToLive the time to live of the cache for the failed
+		 * DNS queries (resolution: seconds)
+		 * @return {@code this}
+		 */
+		NameResolverSpec cacheNegativeTimeToLive(Duration cacheNegativeTimeToLive);
+
+		/**
+		 * Disables the automatic inclusion of an optional record that tries to hint the remote DNS server about
+		 * how much data the resolver can read per response. By default this is enabled.
+		 *
+		 * @param disable true if an optional record is not included
+		 * @return {@code this}
+		 */
+		NameResolverSpec disableOptionalRecord(boolean disable);
+
+		/**
+		 * Specifies whether this resolver has to send a DNS query with the recursion desired (RD) flag set.
+		 * By default this is enabled.
+		 *
+		 * @param disable true if RD flag is not set
+		 * @return {@code this}
+		 */
+		NameResolverSpec disableRecursionDesired(boolean disable);
+
+		/**
+		 * Sets the capacity of the datagram packet buffer (in bytes).
+		 * Default to {@link Build#DEFAULT_MAX_PAYLOAD_SIZE}.
+		 *
+		 * @param maxPayloadSize the capacity of the datagram packet buffer
+		 * @return {@code this}
+		 * @throws IllegalArgumentException if {@code maxPayloadSize} is not positive
+		 */
+		NameResolverSpec maxPayloadSize(int maxPayloadSize);
+
+		/**
+		 * Sets the maximum allowed number of DNS queries to send when resolving a host name.
+		 * Default to {@link Build#DEFAULT_MAX_QUERIES_PER_RESOLVE}.
+		 *
+		 * @param maxQueriesPerResolve the max number of queries
+		 * @return {@code this}
+		 * @throws IllegalArgumentException if {@code maxQueriesPerResolve} is not positive
+		 */
+		NameResolverSpec maxQueriesPerResolve(int maxQueriesPerResolve);
+
+		/**
+		 * Sets the number of dots which must appear in a name before an initial absolute query is made.
+		 * Default to {@link Build#DEFAULT_NDOTS} which determines the value from the OS on Unix
+		 * or uses the value {@code 1}.
```

Comment:
is the default `1` or `-1`? it seems to be `-1` in the concrete spec below.

Reply:
The default is `-1` which means it determines the value from the OS on Unix, if it fails it will be 1.

https://github.com/netty/netty/blob/686df17b1a1bca9ca2d26fe53e5e4eb3404a6b3f/resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java#L145-L152